### PR TITLE
common Accessor: Add access API using std::function

### PR DESCRIPTION
**Pull request by JSUYA**
It can pass user data to the callback function.

https://github.com/Samsung/thorvg/issues/1275

### CODE REVIEW

**Comment by JSUYA** [comment]
I wanted to add an api that uses std::function.
ex)
```cpp
    std::unique_ptr<Picture> access(std::unique_ptr<Picture> picture, std::function<bool(const Paint* paint)> func) noexcept;

```
However, this causes an overloading error with an already opened API that use a function pointer.

File: inc/thorvg.h [hunk 1]
```diff
@@ -1614,6 +1614,19 @@ class TVG_EXPORT Accessor final
...
+     * @brief Access the Picture scene stree nodes.
+     *
+     * @param[in] picture The picture node to traverse the internal scene-tree.
+     * @param[in] func The callback function calling for every paint nodes of the Picture.
+     * @param[in] data Data will be passed to callback function.
+     *
+     * @return Return the given @p picture instance.
+     *
+     * @note The bitmap based picture might not have the scene-tree.
+     */
```
Comment (hermet) [inline inc/thorvg.h:1627]: could you please add @BETA_API tag ?

**Review by hermet — APPROVED** [review]
looks good to me. please check a comment.

**Comment by hermet** [comment]
@JSUYA btw, you suggested this https://github.com/Samsung/thorvg/issues/1275, i agree on this idea. dosen't need to use this type?

**Comment by JSUYA** [comment]
> @JSUYA btw, you suggested this #1275, i agree on this idea. dosen't need to use this type?


As mentioned here(https://github.com/Samsung/thorvg/pull/1277#issuecomment-1319692487), there is an overload issue.

For these two APIs, Example's Accessor.cpp code makes an error.
    std::unique_ptr<Picture> access(std::unique_ptr<Picture> picture, bool(*func)(const Paint* paint)) noexcept;
    std::unique_ptr<Picture> access(std::unique_ptr<Picture> picture, std::function<bool(const Paint* paint)> func) noexcept;

[Accessor example]
```
    auto f = [](const tvg::Paint* paint) -> bool
```
[error]
```
../src/examples/Accessor.cpp:73:5: error: call of overloaded ‘access(std::remove_reference<std::unique_ptr<tvg::Picture>&>::type, tvgDrawCmds(tvg::Canvas*)::<lambda(const tvg::Paint*)>)’ is ambiguous
```

If we allow changing the type of the API that has already been opened, we can use https://github.com/Samsung/thorvg/pull/1280.

**Comment by hermet** [comment]
@JSUYA how about naming the new method "Accessor::set()" then we can deprecate the previous one.

**Comment by hermet** [comment]
great thanks!

**Pull request merged**